### PR TITLE
Creates pre-merge integration test check

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes

This PR adds ability to trigger integration tests via the label "run_integration_test". After changing repo settings this will block PRs from being merged unless the test passes.

## Related issue number (if any)
https://linear.app/aqueducthq/issue/ENG-1024/block-prs-from-merging-unless-integration-tests-pass

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have manually run the integration tests and they are passing.
- [x] All features on the UI continue to work correctly.

### CODE REVIEW

File: .github/workflows/pre-merge-ci.yml
Comment (hsubbaraj-spiral) [inline .github/workflows/pre-merge-ci.yml:9]: I'll need to change this to `main`, it currently won't run because `integration-tests.yml` on `main` doesn't have the `workflow_call` trigger

File: .github/workflows/pre-merge-ci.yml
Comment (kenxu95) [inline .github/workflows/pre-merge-ci.yml:5]: If a PR is labelled, will every push to the PR trigger this? And will I see the action being run on the PR page, or only in the actions tab?

**Review by kenxu95 — APPROVED** [review]
Just have a couple clarifying questions, but feel free to merge after answering!

I remember asking about this but I forgot the answer. Will a PR that is ready to be merged won't be blocked by a more recently failing pre-merge integration tests of another PR?

File: .github/workflows/pre-merge-ci.yml
Comment (hsubbaraj-spiral) [reply .github/workflows/pre-merge-ci.yml:5]: Yes, every push to a PR that is labeled will trigger a run of the integration test. If you know you are going to push multiple commits and don't want to run the tests, you can remove the label, push your commits, then add the label back to pass the tests before merging.

**Comment by hsubbaraj-spiral** [comment]
> Just have a couple clarifying questions, but feel free to merge after answering!
> 
> I remember asking about this but I forgot the answer. Will a PR that is ready to be merged won't be blocked by a more recently failing pre-merge integration tests of another PR?

Tests that are triggered for different PRs are independent of each other (run using the PR branch code). So no, you should not be blocked by a different non-merged PR failing its tests.

**Comment by hsubbaraj-spiral** [comment]
@kenxu95 one more question, how does this affect the reporting to the #github-notifications channel on Slack? I don't imagine that we want to be notified every time someone runs their tests on a PR.

**Comment by kenxu95** [comment]
yeah um, if we're sure there is no way to merge a PR without the integration tests running and passing on the latest version of that PR, the slack notification should no longer be necessary.

**Pull request merged**
